### PR TITLE
Move hiding series to PanelContext and hide series from PieChart

### LINKED ISSUE #31116: PieChart: New core plugin / visualization
* [x] Basic display & options 
* [x] Legend below or to the right
* [x] Legend stats percent and or value, think there needs to be handled a bit differently than the legend stats in the graph as the panel already has reduced values so these are not exactly the same 
* [x] Change color from legend (saved as override) 
* [x] Data links support 
  * [x] Bug, data links do not work: https://github.com/grafana/grafana/issues/32637
* [x] Add migration for [PieChart](https://grafana.com/grafana/plugins/grafana-piechart-panel) -> PieChart NG
* [x] Add e2e test (#33737)
* [x] Hover over legend to highlight pie piece (needs a big change in legend component) (#32941)
* [x] Move DataFrame -> FieldDisplay values into PieChart componet so PieChart component takes in data: DataFrame[] prop  instead of fieldDisplayValues 
* [x] Unify Tooltip to look the way it looks in TimeSeries (preferably use the same components) #33032
* [ ] Legend & tooltip - unify panel options creation between timeseries & piechart ([addLegenOptions](https://github.com/grafana/grafana/blob/45eaf3cfdc733f294ec1f2dc954175eeda30ef0e/public/app/plugins/panel/timeseries/config.ts#L283), addTooltipOptions)
* [ ] Change the hover & highlight to one down the other slices (using opacity) (#33778)

|   |   |
|---|---|
| image  |  image |

* [ ] Make it possible to toggle series on/off via legend (#33756) (see how it's done in time series panel: https://github.com/grafana/grafana/blob/master/public/app/plugins/panel/timeseries/TimeSeriesPanel.tsx#L28)
* [x] Fix storybook (#33696)

For it to look good by default we need a new color palette! (using the named colors), planed for v8

### PR DESCRIPTION
**Pull request by oscarkilhed**
**What this PR does / why we need it**:
This moves the hiding of series to the PanelContext in order to make it easier to reuse this functionality across multiple panels. It then uses the PanelContext to hide series in the PieChart.

**Special notes for your reviewer**:
There is some duplication of the hide series because I didn't want to risk breaking the `ExploreGraphNGPanel` at this point

### CODE REVIEW

File: packages/grafana-ui/src/components/PanelChrome/PanelContext.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { EventBusSrv, EventBus } from '@grafana/data';
 import React from 'react';
+import { GraphNGLegendEventMode } from '../GraphNG/types';
```
Comment (ryantxu) [inline packages/grafana-ui/src/components/PanelChrome/PanelContext.ts:3]: Can we move this type to PanelChrome?

Comment (oscarkilhed) [reply packages/grafana-ui/src/components/PanelChrome/PanelContext.ts:3]: Ah yes, I was gonna get around to that but forgot :)

**Comment by ryantxu** [comment]
In general looks great!  simplifying these complex behavior<>field overrides will help a lot.

We need to move around some of the config options so they make sense -- currenty tied to GraphNG is not quite right :)

File: packages/grafana-ui/src/components/VizLegend/VizLegend.tsx [hunk 2]
```diff
@@ -13,13 +15,14 @@ export const VizLegend: React.FunctionComponent<LegendProps> = ({
   items,
   displayMode,
   sortBy: sortKey,
+  disableSeriesIsolation,
```
Comment (dprokop) [inline packages/grafana-ui/src/components/VizLegend/VizLegend.tsx:18]: nit - not sure about this name, maybe `isolateSeriesOnToggle`? naming is hard 🤮

File: packages/grafana-ui/src/components/PanelChrome/PanelContext.ts (hunk 1 above)
Comment (dprokop) [reply packages/grafana-ui/src/components/PanelChrome/PanelContext.ts:3]: And I think we can also rename it to sth GraphNG agnostic. `SeriesToggleMode`?

**Review by dprokop — COMMENTED** [review]
Great, this is exactly the direction we talked about :)

**Comment by ryantxu** [comment]
This looks great -- I would like to merge soon (today) so that other PRs get smaller and easier to manage :)

**Review by ryantxu — APPROVED** [review]

**Pull request merged**